glogger/handler.py

```diff
@@ -51,21 +51,31 @@ def __init__(
         :param overflow_drop_factor: Percentage of messages to be dropped when buffer becomes full.
         """
         super().__init__(logging.DEBUG)
-        self.sender = sender
         self.max_message_size = max_message_size  # maximum message size
 
         self.stdout_logger = get_stdout_logger()
         self.jsonify = JSONEncoder(separators=(",", ":"), default=repr).encode  # compact, no whitespace
         self.messages_buffer = MessagesBuffer(max_total_length, overflow_drop_factor)
         self.messages_buffer.head_serial_no = continue_from
 
+        self.sender: Optional[Sender] = None
+        self.init_sender(sender)
+
+    def init_sender(self, sender: Optional[Sender]) -> None:
```

Comment:
Actually, let's accept a non-optional `Sender` here and check if it's None in the __init__ function instead